### ISSUE TITLE
Unable to open pdf file from another application

### ISSUE THREAD

**Issue by panbroggi** [koreader/koreader#8484]
* KOReader version: 2021.11
* Device: Samsung Galaxy Tab s6

#### Issue
I cannot open a file from the Zoo for Zotero app.
It stores the files in `/sdcard/Android/data/com.mickstarify.zooforzotero/cache`.

#### Steps to reproduce
Download an attachment to your device in Zoo for Zotero. Tap on it and choose open with > KOReader 

##### `crash.log`
attached
[crash.log](https://github.com/koreader/koreader/files/7607704/crash.log)

**Comment by pazos** [comment]
Hi, that's a private dir of the `com.mickstarify.zooforzotero` package. Since android 10/11? only that package is able to read that directory.

**Comment by panbroggi** [comment]
I see. But why other apps like Librera or Xodo can open these files?

**Comment by pazos** [comment]
> I see. But why other apps like Librera or Xodo can open these files?

They make a copy of each single book you open in them. We don't want that behaviour.

We open books directly, no matter where they're placed. The only exception is in `Android/data`, these folders are private.

**Comment by panbroggi** [comment]
I am quite sure they modify the original document, since the modification is detected and the file synced by the app. Do they copy it back to the original location?

**Comment by pazos** [comment]
They can't read or write to `/sdcard/Android/data/com.mickstarify.zooforzotero/cache`

The app that passes the intent could use a FileProvider and grant read&write permissions on that document specific uri. From our PoV it doesn't matter, as the file is not available using the filesystem api.

Most apps use SAF to let you to choose a path to store documents. In these cases everything works because these paths are public and we can usually read&write there (unless they're placed on a external sdcard/ USB OTG, where we can read but not write).

TL;DR: we require filesystem access to files. Use another reader app if you want to use Zoo for Zotero or convince the developer to store documents in user designed path instead of private paths, so we can read and write there.

**Comment by panbroggi** [comment]
Thank you for explaining me the process, it was instructive.

**Comment by pazos** [comment]
> Thank you for explaining me the process, it was instructive.

You're welcome. 

The only thing we can do is to detect a private path and prompt the user to import the file. Not sure if that's an improvement because the user might have some expectations about the file autosynced back to the app that was used to open the document.
